tests/end2end/fixtures/test_qutewm.py

```diff
@@ -0,0 +1,93 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2016 Daniel Schadt
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests for qutewm and the qutewm fixtures."""
+
+from PyQt5.QtWidgets import QMainWindow
+from PyQt5.QtCore import QEvent, pyqtSignal
+
+import pytest
+
+from qutebrowser.utils import qtutils
+from end2end.fixtures import qutewm
+
+
+class Window(QMainWindow):
+
+    """A window class suitable for tests.
+
+    This window provides a activated signal.
+    """
+
+    activated = pyqtSignal()
+
+    def changeEvent(self, ev):
+        if ev.type() == QEvent.ActivationChange and self.isActiveWindow():
+            self.activated.emit()
+
+
+@pytest.mark.qutewm
+class TestQuteWM:
+
+    @classmethod
+    def get_window(cls):
+        """Return a window suitable for testing."""
+        window = Window()
+        return window
```

Comment:
Old remnants when I had more set-up to a window when I tried to get the "Activation changed" signal.